TechnicalDocuments/RefactorAppMainUserSession.md

```diff
@@ -0,0 +1,38 @@
+
+Needless to say this a rather complicated entity with poor separation of concerns. `AppMainUserSession` was designed in this way to work around a number of legacy problems, among them multiple authentication domains and that version 2 of the Babylon app relied heavily on singletons. It is, however, doing its job reasonably well and would probably not be marked for refactoring if it was not for the fact that it blocks moving our app away from accessing business controllers directly and instead use the APIs in the Babylon SDK.
+
+More background information about what `MainUserSession` is and what we would like it to become can be found in this [deck](https://drive.google.com/drive/u/0/folders/1PVm5ztVYGpOdqLZxTmIl2BXff8ux8MnK), it is a Keynote presentation so you need download and unzip it.
+
+Proposed refactoring is to change AppMainUserSession from being a singular to a composite entity. It will still comply with the current set of protocols, but implement them by forwarding to the relevant component. This can, hopefully, be done in a step by step manner to avoid a high risk gargantuan pull request.
+
+# Step One, Move to the Babylon Main Project.
+
+Both the SessionViewModel, MainUserSession and AppMainUserSession are specific to the Babylon app targets and not needed to implement the SDK. They should thus be moved out of BabylonCore. To move them to the Octopus folder in the main project seems logical, but BabylonDependencies is also technically possible. Moving them to the main project will require minor refactoring to a couple of UI builders that consume MainUserSession directly.
+
+# Step Two, Add an `AuthenticatedContext` Instance.
+
+Serving an `AuthenticatedContext` through the `MainUserSession` interface will allow builders to create instances of all SDK APIs. It should be noted that all our SDK APIs are very lightweight both in terms of memory consumption and time to create. There is thus no need to serve them centrally unless they hold state that should be shared.
+
+Methods in `MainUserSession` that implement requirements for `AuthenticatedAccessible` will be changed to route through the `AuthenticatedContext`.
+
+The `AuthenticatedContext` can be created by the `SessionViewModel` until sign in, sign up and the app boot strap has been refactored to employ the `Auth` SDK.
+
+# Step Three, Add a `ClinicalRecords` Instance.
```

Comment:
Before this step I see another step"
# Replace `session` from SessionViewModel and from AppMainUSerSession with Auth instance.

First the SessionViewModel should use the `Auth.login` methods and and use the Authenticated context provided through that flow.

Reply:
Can we do this without refactoring sign in and sign up?